### PR TITLE
Enable checkstyle

### PR DESCRIPTION
**Pull request by chrisalice**
Enable checkstyle with a modified copy of [the Auto project's checkstyle rules], and fix violations.

Alterations:
 * Warnings suppressed by the comment `// Autogenerated code. Do not modify.`
 * Length checks disabled for test classes named `.*SourceTest`
 * No [cache file](http://checkstyle.sourceforge.net/config.html#Checker)
 * Underscores allowed in test method names
 * Empty types allowed
 * RedundantThrows removed as it's [no longer in CheckStyle](https://github.com/checkstyle/checkstyle/issues/473)

[the Auto project's checkstyle rules]: https://github.com/google/auto/blob/0c06a2345f71f053714d37bb6549d3460c999f2d/checkstyle.xml

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+
+<!--
+This file is based on
+https://github.com/google/auto/blob/0c06a2345f71f053714d37bb6549d3460c999f2d/checkstyle.xml
```
Comment (hdurer) [inline config/checkstyle/checkstyle.xml]: That's a github page with the contents you want.
Not sure if that is better or worse than the content we want directly, i.e. something like https://raw.githubusercontent.com/google/auto/0c06a2345f71f053714d37bb6549d3460c999f2d/checkstyle.xml

Comment (chrisalice) [inline config/checkstyle/checkstyle.xml]: I decided this one was better as it lets you access more context

Comment (hdurer) [inline config/checkstyle/checkstyle.xml]: True.  Makes sense.

**Comment by hdurer** [comment]
LGTM

**Pull request merged**